openwisp_controller/config/base/device.py

```diff
@@ -299,7 +308,8 @@ def _check_changed_fields(self):
         self._get_initial_values_for_checked_fields()
         # Execute method for checked for each field in self._changed_checked_fields
         for field in self._changed_checked_fields:
-            getattr(self, f'_check_{field}_changed')()
+            if hasattr(self, f'_check_{field}_changed'):
+                getattr(self, f'_check_{field}_changed')()
```

Comment:
we can check via callable and use early exit:

```suggestion
            method = getattr(self, f'_check_{field}_changed', None)
            if not callable(method):
            	continue
            method()
```